app/components/primer/icon_button.rb

```diff
@@ -54,11 +75,22 @@ def initialize(icon:, scheme: DEFAULT_SCHEME, box: false, **system_arguments)
       )
 
       validate_aria_label
-    end
 
-    def call
-      render(Primer::BaseButton.new(**@system_arguments)) do
-        render(Primer::OcticonComponent.new(icon: @icon))
+      @aria_label = aria("label", @system_arguments)
+      @aria_description = aria("description", @system_arguments)
+
+      # If we have both an `aria-label` and a `aria-description`, we create a `Tooltip` with the description type and keep the `aria-label` in the button.
+      # Otherwise, the `aria-label` is used as the tooltip text, which is the `aria-labelled-by` of the button, so we don't set it in the button.
+      @tooltip_type = @aria_label.present? && @aria_description.present? ? :description : :label
+
+      if @tooltip_type == :description
+        @system_arguments.delete(:"aria-description")
+        @system_arguments[:aria].delete(:description) if @system_arguments.include?(:aria)
+        @tooltip_text = @aria_description
+      else
+        @system_arguments.delete(:"aria-label")
+        @system_arguments[:aria].delete(:label) if @system_arguments.include?(:aria)
+        @tooltip_text = @aria_label
       end
```

Comment:
```suggestion
      @tooltip_arguments = {
        for_id: @system_arguments[:id],
        direction: tooltip_direction
      }

      # If we have both an `aria-label` and a `aria-description`, we create a `Tooltip` with the description type and keep the `aria-label` in the button.
      # Otherwise, the `aria-label` is used as the tooltip text, which is the `aria-labelled-by` of the button, so we don't set it in the button.
      if @aria_label.present? && @aria_description.present?
        @system_arguments.delete(:"aria-description")
        @system_arguments[:aria].delete(:description) if @system_arguments.include?(:aria)
        @tooltip_arguments[:text] = @aria_description
        @tooltip_arguments[:type] = :description
      else
        @system_arguments.delete(:"aria-label")
        @system_arguments[:aria].delete(:label) if @system_arguments.include?(:aria)
        @tooltip_arguments[:text] = @aria_label
        @tooltip_arguments[:type] = :label
      end
```